src/kernels/jupyter/session/jupyterRequestCreator.web.ts

```diff
@@ -16,63 +16,50 @@ const JupyterWebSockets = new Map<string, WebSocketIsomorphic & IKernelSocket>()
 /* eslint-disable @typescript-eslint/no-explicit-any */
 @injectable()
 export class JupyterRequestCreator implements IJupyterRequestCreator {
-    public getRequestCtor(getAuthHeader?: () => any) {
+    public getRequestCtor(cookieString?: string, allowUnauthorized?: boolean, getAuthHeaders?: () => any) {
         class AuthorizingRequest extends Request {
             constructor(input: RequestInfo, init?: RequestInit) {
                 super(input, init);
 
                 // Add all of the authorization parts onto the headers.
                 const origHeaders = this.headers;
-                const authorizationHeader = getAuthHeader!();
-                const keys = Object.keys(authorizationHeader);
-                keys.forEach((k) => origHeaders.append(k, authorizationHeader[k].toString()));
-                origHeaders.set('Content-Type', 'application/json');
-
-                // Rewrite the 'append' method for the headers to disallow 'authorization' after this point
-                const origAppend = origHeaders.append.bind(origHeaders);
-                origHeaders.append = (k, v) => {
-                    if (k.toLowerCase() !== 'authorization') {
-                        origAppend(k, v);
-                    }
-                };
+
+                if (getAuthHeaders) {
+                    const authorizationHeader = getAuthHeaders();
+                    const keys = Object.keys(authorizationHeader);
+                    keys.forEach((k) => origHeaders.append(k, authorizationHeader[k].toString()));
+                    origHeaders.set('Content-Type', 'application/json');
+
+                    // Rewrite the 'append' method for the headers to disallow 'authorization' after this point
+                    const origAppend = origHeaders.append.bind(origHeaders);
+                    origHeaders.append = (k, v) => {
+                        if (k.toLowerCase() !== 'authorization') {
+                            origAppend(k, v);
+                        }
+                    };
+                }
+
+                // Append the other settings we might need too
+                if (allowUnauthorized) {
+                    // rejectUnauthorized not allowed in web so we can't do anything here.
+                }
+
+                if (cookieString) {
```

Comment:
I don't have a good way to test this. This would be necessary for Azure ML support in the web. Might need this for github depending upon if the VMs they create have anything in between them and the user.